### PR TITLE
Location are working

### PR DESCRIPTION
**Pull request by leobenkel**

### CODE REVIEW

File: js/utility/read-file.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ define([
     var readFileAsJson = (path) => {
         return readFile(path)
         .then((data) => {
+            if (data !== null && typeof data === 'object' && typeof data !== 'string') {
+                return data;
+            }
             return JSON.parse(data);
```
Comment (jacklehamster) [inline js/utility/read-file.js:24]: Also try catch, in case the json is not valid

File: js/app/location.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+                wasDrawn = false;
+                return Promise.resolve(returnedData);
+            }
+        }
+
+        this.render = (renderer, mouse) => {
+            return handleUpdate(renderer, mouse);
+        }
+
+        this.handleCursorMove = (renderer, mouse) => {
```
Comment (jacklehamster) [inline js/app/location.js:134]: Why can't you just write this.handleCursorMove = handleUpdate?

File: js/app/location.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
+define([
+    '../utility/check-data.js',
+],
+(CheckData) => {
+    var Location = function(parent, data) {
+        var myself = self;
```
Comment (jacklehamster) [inline js/app/location.js:6]: Why is this needed? Can't you just use 'self'?

File: js/utility/read-file.js (hunk 1 above)
Comment (leobenkel) [reply js/utility/read-file.js:24]: if the json is not valid it cannot run, no need to fail safely.

File: js/app/location.js (hunk 3 above)
Comment (leobenkel) [reply js/app/location.js:6]: I added myself everywhere. so it all looks like the same class pattern but it is not used much.

File: js/app/location.js (hunk 2 above)
Comment (leobenkel) [reply js/app/location.js:134]: in case i want to add more code later :)

**Review by jacklehamster — APPROVED** [review]
